api/api/V1/Device.ts

```diff
@@ -699,4 +699,30 @@ export default function (app: Application, baseUrl: string) {
       });
     }
   );
+
+  /**
+     * @api {post} /api/v1/devices/heartbeat send device heartbeat
+     * @apiName heartbeat
+     * @apiGroup Device
+     *
+     * @apiUse V1DeviceAuthorizationHeader
+     *
+     * @apiBody {Date} nextHeartbeat time next heart beat is expected
+
+     * @apiUse V1ResponseSuccess
+     * @apiUse V1ResponseError
+     */
+  app.post(
+    `${apiUrl}/heartbeat`,
+    extractJwtAuthorisedDevice,
```

Comment:
This won't actually give you a real `Device` model object, just the id, so you will want to do `const actualDevice = await models.Device.getByPk(response.locals.requestDevice.id)` before you try and call `updateHeartbeat()` on it.